File: code_llm/zero-shot/gemini-3.1-pro-preview/Retas.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

struct Line {
    long long y1, y2;
    bool operator<(const Line& other) const {
        if (y1 != other.y1)
            return y1 < other.y1;
        return y2 > other.y2;
    }
};

long long merge_sort(vector<long long>& arr, vector<long long>& temp, int l, int r) {
    if (l >= r) return 0;
    int mid = l + (r - l) / 2;
    long long inv = merge_sort(arr, temp, l, mid) + merge_sort(arr, temp, mid + 1, r);
    
    int i = l, j = mid + 1, k = l;
    
    while (i <= mid && j <= r) {
        if (arr[i] < arr[j]) {
            temp[k++] = arr[i++];
        } else {
            inv += (mid - i + 1);
            temp[k++] = arr[j++];
        }
    }
    while (i <= mid) temp[k++] = arr[i++];
    while (j <= r) temp[k++] = arr[j++];
    
    for (int p = l; p <= r; p++) {
        arr[p] = temp[p];
    }
    return inv;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int n;
    long long x1, x2;
    if (!(cin >> n >> x1 >> x2)) return 0;
    
    vector<Line> lines(n);
    for (int i = 0; i < n; i++) {
        long long a, b;
        cin >> a >> b;
        lines[i].y1 = a * x1 + b;
        lines[i].y2 = a * x2 + b;
    }
    
    sort(lines.begin(), lines.end());
    
    vector<long long> y2(n);
    for (int i = 0; i < n; i++) {
        y2[i] = lines[i].y2;
    }
    
    vector<long long> temp(n);
    long long ans = merge_sort(y2, temp, 0, n - 1);
    
    cout << ans << "\n";
    
    return 0;
}

File: output/csbc_semish_2026/x_ai_grok_code_fast_1/few/python/Retas.py
import sys

data = sys.stdin.read().split()
idx = 0
N = int(data[idx]); idx += 1
X1 = int(data[idx]); idx += 1
X2 = int(data[idx]); idx += 1
lines = []
for i in range(N):
    A = int(data[idx]); idx += 1
    B = int(data[idx]); idx += 1
    lines.append((A, B))

if X1 == X2:
    from collections import Counter
    ys = [A * X1 + B for A, B in lines]
    cnt = Counter(ys)
    ans = 0
    for v in cnt.values():
        ans += v * (v - 1) // 2
    print(ans)
else:
    pairs = [(A * X1 + B, A * X2 + B) for A, B in lines]
    pairs.sort(key=lambda x: x[0])
    
    def merge(arr, left, mid, right):
        n1 = mid - left + 1
        n2 = right - mid
        L = [arr[left + i] for i in range(n1)]
        R = [arr[mid + 1 + j] for j in range(n2)]
        i = 0
        j = 0
        k = left
        inv = 0
        while i < n1 and j < n2:
            if L[i][1] <= R[j][1]:
                arr[k] = L[i]
                i += 1
            else:
                arr[k] = R[j]
                j += 1
                inv += (n1 - i)
            k += 1
        while i < n1:
            arr[k] = L[i]
            i += 1
            k += 1
        while j < n2:
            arr[k] = R[j]
            j += 1
            k += 1
        return inv
    
    def merge_sort(arr, left, right):
        inv = 0
        if left < right:
            mid = (left + right) // 2
            inv += merge_sort(arr, left, mid)
            inv += merge_sort(arr, mid + 1, right)
            inv += merge(arr, left, mid, right)
        return inv
    
    ans = merge_sort(pairs, 0, len(pairs) - 1)
    print(ans)